sprockets/mixins/amqp/amqp.py

```diff
@@ -0,0 +1,203 @@
+        username: tinman
+        password: tornado
+
+"""
+import os
+from tornado import gen, locks
+import logging
+import pika
+from pika.adapters import tornado_connection
+from tornado import web
+
+LOGGER = logging.getLogger(__name__)
+
+# a datetime.timedelta relative to the current time
+CHANNEL = None
+CONNECTION = None
+WAIT_TIMEOUT = os.environ.get('WAIT_TIMEOUT')
+
+
+class RabbitMQRequestMixin(web.RequestHandler):
+    """The request handler will connect to RabbitMQ on the first request,
+    blocking until the connection and channel are established. If RabbitMQ
+    closes it's connection to the app at any point, a connection attempt will
+    be made on the next request.
+
+    Expects the :envvar:`AMQP` environment variable to construct
+    :class:`pika.connection.URLParameters`.
+
+    https://github.com/gmr/tinman/blob/master/tinman/handlers/rabbitmq.py
+
+    """
+
+    def initialize(self):
+        super(RabbitMQRequestMixin, self).initialize()
+        self.amqp_conn_condition = locks.Condition()
+        self.amqp_channel_condition = locks.Condition()
+
+    def _connect_to_rabbitmq(self):
+        """Connect to RabbitMQ and assign a class attribute"""
+        global CONNECTION
+        LOGGER.info('Creating a new RabbitMQ connection')
+        CONNECTION = self._new_rabbitmq_connection()
+
+    def _new_message_properties(self, content_type=None, content_encoding=None,
+                                headers=None, delivery_mode=None,
+                                priority=None, correlation_id=None,
+                                reply_to=None, expiration=None,
+                                message_id=None, timestamp=None,
+                                message_type=None, user_id=None, app_id=None):
+        """Create a BasicProperties object, with the properties specified
+
+        :param str content_type: MIME content type
+        :param str content_encoding: MIME content encoding
+        :param dict headers: Message header field table
+        :param int delivery_mode: Non-persistent (1) or persistent (2)
+        :param int priority: Message priority, 0 to 9
+        :param str correlation_id: Application correlation identifier
+        :param str reply_to: Address to reply to
+        :param str expiration: Message expiration specification
+        :param str message_id: Application message identifier
+        :param int timestamp: Message timestamp
+        :param str message_type: Message type name
+        :param str user_id: Creating user id
+        :param str app_id: Creating application id
+        :rtype: pika.BasicProperties
+
+        """
+        return pika.BasicProperties(content_type, content_encoding, headers,
+                                    delivery_mode, priority, correlation_id,
+                                    reply_to, expiration, message_id,
+                                    timestamp, message_type, user_id, app_id)
+
+    def _new_rabbitmq_connection(self):
```

Comment:
Is there any value in having a one line function that simply calls another function?  Seems like 10 or so lines of fat to my eyes.